packages/protocol/contracts/common/FeeHandler.sol

```diff
@@ -29,19 +41,33 @@ contract FeeHandler is Ownable, Initializable, UsingRegistry, ICeloVersionedCont
 
   // Min units that can be burned
   uint256 public constant MIN_BURN = 200;
-  // Historical amounts burned by this contract
-  mapping(address => uint256) public pastBurn;
-  // Max amounts that can be burned in a day for a token
-  mapping(address => uint256) public dailyBurnLimit;
-  // Max amounts that can be burned today for a token
-  mapping(address => uint256) public currentDayLimit;
+
   // last day the daily limits were updated
   uint256 public lastLimitDay;
-  // router addresses that can be set for a token
-  mapping(address => address[]) public routerAddresses;
 
-  // Max slippage that can be accepted when burning a token
-  mapping(address => FixidityLib.Fraction) public maxSlippage;
+  FixidityLib.Fraction public burnFraction; // 80%
+
+  address public feeBeneficiary;
+
+  // This mapping can not be public because it contains  a FixidityLib.Fraction
+  // and that'd be only supported with experimental features in this
+  // compiler version
+  mapping(address => TokenState) private tokenStates;
+
+  struct TokenState {
+    address handler;
+    bool active;
+    FixidityLib.Fraction maxSlippage;
+    // Max amounts that can be burned in a day for a token
+    uint256 dailyBurnLimit;
+    // Max amounts that can be burned today for a token
+    uint256 currentDayLimit;
+    uint256 toDistribute;
+    // Historical amounts burned by this contract
+    uint256 pastBurn;
+  }
+
+  address[] public activeTokens;
```

Comment:
We could use AddressSet from OpenZeppelin it would make sure that there are no duplicates and Search and Removal are both O(1) instead of O(n) for array

Reply:
great idea, will fix after merge